### PR TITLE
Suisin/upm826/success modal phone number verification

### PR DESCRIPTION
**Pull request by suisin-deriv**
## Changes:

Add success modal for phone number verification 

### Screenshots:

Screenshot 2024-05-09 at 6 11 14 PM

### CODE REVIEW

File: packages/stores/types.ts [hunk 1]
```diff
@@ -744,6 +744,8 @@ type TUiStore = {
     setRealAccountSignupEnd: (status: boolean) => void;
     setPurchaseState: (index: number) => void;
     simple_duration_unit: string;
+    should_show_phone_number_verified_modal: boolean;
```
Comment (amina-deriv) [inline packages/stores/types.ts:747]: This modal will be triggered only when phone number otp is verified. Isnt?
Can we just trigger the modal on success on API call then?
do we need a observer for it

Comment (suisin-deriv) [reply packages/stores/types.ts:747]: Yeap but the modal will be displayed on personal-details-page, so when we check for verified: 1 api call this modal will show up everytime isn't 🤔 I was thinking that when user clicks on `verify` button, when the API call isSuccess then we put the value in store and redirect user back to personal details then dispaly the successful modal

Comment (amina-deriv) [reply packages/stores/types.ts:747]: Lets check with design team on this flow

**Review by utkarsha-deriv — APPROVED** [review]

File: packages/account/src/Sections/Profile/PhoneVerification/__test__/phone-number-verified-modal.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    };
+
+    it('it should render PhoneNumberVerifiedModal', () => {
+        (React.useState as jest.Mock).mockReturnValue([true, jest.fn()]);
+        renderModal();
+        expect(screen.getByText(/Verification successful/)).toBeInTheDocument();
+        expect(screen.getByText(/That's it! Your number is verified./)).toBeInTheDocument();
+    });
+
+    it('it should render setShouldShowPhoneNumberVerifiedModal when done is clicked', () => {
```
Comment (amina-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/__test__/phone-number-verified-modal.spec.tsx:50]: it should close PhoneNumberVerifiedModal and navigate to PersonalDetails section  when done is clicked

**Review by amina-deriv — APPROVED** [review]

**Pull request merged**